src/pages/communication/components/chat/ChatContainer.tsx

```
@@ -1,21 +1,90 @@
+  const isReplyTyping = draftReply.length > 0;
+  const isListeningTyping = draftListening.length > 0;
+
+  // 마지막으로 입력해서 위로 올라간 쪽을 기억해뒀다가, 양쪽 다 비어있는(idle) 동안에도
+  // 그 순서를 그대로 유지한다. 처음(둘 다 입력한 적 없음)엔 오른쪽이 위인 기본값을 쓴다.
+  const [lastActiveSide, setLastActiveSide] = useState<'left' | 'right'>(
+    'right',
+  );
+
+  useEffect(() => {
+    if (isReplyTyping) {
+      setLastActiveSide('right');
```

ESLint

[error] Error: Calling setState synchronously within an effect can trigger cascading renders

Effects are intended to synchronize state between React and external systems such as manually updating the DOM, state management libraries, or other platform APIs. In general, the body of an effect should do one or both of the following:
* Update external systems with the latest state from React.
* Subscribe for updates from some external system, calling setState in a callback function when external state changes.

Calling setState synchronously within an effect body causes cascading renders that can hurt performance, and is not recommended. (https://react.dev/learn/you-might-not-need-an-effect).

/home/jailuser/git/src/pages/communication/components/chat/ChatContainer.tsx:42:7
  40 |   useEffect(() => {
  41 |     if (isReplyTyping) {
> 42 |       setLastActiveSide('right');
     |       ^^^^^^^^^^^^^^^^^ Avoid calling setState() directly within an effect
  43 |     } else if (isListeningTyping) {
  44 |       setLastActiveSide('left');
  45 |     }

(react-hooks/set-state-in-effect)